src/App.jsx

```diff
@@ -1,63 +1,162 @@
-import { initialColors } from "./lib/colors";
-import Color from "./Components/Color/Color";
-import ColorForm from "./Components/ColorForm/ColorForm";
+import { initialThemes } from "./lib/colors";
+import Theme from "./Components/Theme/Theme";
+import ThemeSelect from "./Components/ThemeSelect/ThemeSelect";
 import useLocalStorageState from "use-local-storage-state";
 import { uid } from "uid";
 import "./App.css";
 
 function App() {
-  const [colors, setColors] = useLocalStorageState("themeColors", {
-    defaultValue: initialColors,
+  const [themes, setThemes] = useLocalStorageState("themes", {
+    defaultValue: initialThemes,
   });
 
+  const [currentTheme, setCurrentTheme] = useLocalStorageState("currentTheme", {
+    defaultValue: initialThemes[0],
+  });
+
+  // console.log("themes: ", themes);
+  // console.log("currentTheme: ", currentTheme);
+  // console.log("currentThemeColors: ", currentTheme.colors);
+
+  function handleThemeChange(event) {
+    const themeName = event.target.value;
+    const filteredTheme = themes.filter((theme) => theme.name === themeName);
+    setCurrentTheme(filteredTheme[0]);
+  }
+
   function handleAddColor(data) {
     const newColor = {
       id: uid(6),
       role: data.role,
       hex: data.hex,
       contrastText: data.contrastText,
     };
-    setColors([newColor, ...colors]);
+
+    const updatedThemes = themes.map((theme) => {
+      if (theme.name === currentTheme.name) {
+        return {
+          ...theme,
+          colors: [newColor, ...theme.colors],
+        };
+      }
+      return theme;
+    });
+    setThemes(updatedThemes);
+
+    const updatedCurrentTheme = updatedThemes.find(
+      (theme) => theme.name === currentTheme.name
+    );
+    setCurrentTheme(updatedCurrentTheme);
   }
 
   function handleDeleteColor(id) {
-    setColors(colors.filter((color) => color.id !== id));
+    console.log("DELETE");
+
+    const updatedThemes = themes.map((theme) => {
+      if ((theme.name = currentTheme.name)) {
```

Comment:
Corrected the assignment operator `=` to strikt equal comparison `===`

```suggestion
      if ((theme.name === currentTheme.name)) {
```